### PR TITLE
Detect operator-sdk local up

### PR DESCRIPTION
**Pull request by wanghaoran1988**
This is a same one with https://github.com/openshift/certman-operator/pull/88

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -146,10 +150,17 @@ func start() error {
...
+	// detect if operator-sdk up local is run
+	detectSDKLocal := os.Getenv(envSDK)
+
+	// Configure metrics. If it errors, log the error and exit.
+	if detectSDKLocal == "local" {
+		log.Info("Skipping metrics configuration; not running in a cluster.")
+	} else {
+		if err := metrics.ConfigureMetrics(context.TODO(), *metricsServer); err != nil {
+			log.Error(err, "Failed to configure Metrics")
+			os.Exit(1)
```
Comment (mjudeikis) [inline cmd/manager/main.go]: return err

Comment (mjudeikis) [reply cmd/manager/main.go]: and replace top ones too to `returun err`

File: cmd/manager/main.go [hunk 2]
```diff
@@ -146,10 +150,17 @@ func start() error {
...
 
-	// Configure metrics if it errors log the error but continue
-	if err := metrics.ConfigureMetrics(context.TODO(), *metricsServer); err != nil {
-		log.Error(err, "Failed to configure Metrics")
-		os.Exit(1)
+	// detect if operator-sdk up local is run
+	detectSDKLocal := os.Getenv(envSDK)
+
+	// Configure metrics. If it errors, log the error and exit.
+	if detectSDKLocal == "local" {
```
Comment (mjudeikis) [inline cmd/manager/main.go]: `if os.Getenv(envSDK) == "local"`

Comment (wanghaoran1988) [reply cmd/manager/main.go]: @mjudeikis Done

File: cmd/manager/main.go (hunk 1 above)
Comment (wanghaoran1988) [reply cmd/manager/main.go]: @mjudeikis Do you mean replace all the `os.Exit(1)` to `return err` in this `start()` func?

Comment (mjudeikis) [reply cmd/manager/main.go:160]: yes

**Pull request closed without merging**